File: binarySearch/binarySearch.zig
const std = @import("std");
const print = std.debug.print;

pub fn binarySearch(items: []u8, target: u8) ?usize {
    var left: usize = 0;
    var right = items.len;

    while (left < right) {
        const mid = left + (right - left) / 2;
        if (items[mid] == target) {
            return mid;
        } else if (items[mid] < target) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    return null;
}

pub fn main() anyerror!void {
    var items = [_]u8{ 2, 4, 6, 8, 10 };
    const target: u8 = 122;
    const resp = binarySearch(items[0..], target);

    if (resp == null) {
        print("The number {any}, was not found in the array", .{target});
    } else {
        print("Index: {any}", .{resp});
    }
}

File: binarySearch/binarySearch.c
#include <stdio.h>
#include <math.h>

int binarySearch(int arr[], int size, int target) {
    int left = 0;
    int right = size - 1;

    while (left <= right)
    {
        int mid = floor((left + right) / 2);
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    
    return -1;
}

int main() {
    int arr[] = {2, 4, 6, 8, 10};
    int size = sizeof(arr) / sizeof(arr[0]);
    int target = 6;
    int res = binarySearch(arr, size, target);
    printf("Index: %d\n", res);
    return 0;
}
